### PR TITLE
[ROCm] Supports ROCm6.0 reorganization and cleanup

### PR DESCRIPTION
**Pull request by BLOrange-AMD**
cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### CODE REVIEW

File: torch/utils/hipify/cuda_to_hip_mappings.py [hunk 1]
```diff
@@ -606,8 +609,8 @@
         ),
         ("cooperative_groups.h", ("hip/hip_cooperative_groups.h", CONV_INCLUDE, API_RUNTIME)),
         ("vector_types.h", ("hip/hip_vector_types.h", CONV_INCLUDE, API_RUNTIME)),
-        ("cublas.h", ("hipblas/hipblas.h", CONV_INCLUDE_CUDA_MAIN_H, API_BLAS)),
-        ("cublas_v2.h", ("hipblas/hipblas.h", CONV_INCLUDE_CUDA_MAIN_H, API_BLAS)),
+        ("cublas.h", ("rocblas/rocblas.h", CONV_INCLUDE_CUDA_MAIN_H, API_BLAS)),
+        ("cublas_v2.h", ("rocblas/rocblas.h", CONV_INCLUDE_CUDA_MAIN_H, API_BLAS)),
```
Comment (jeffdaily) [inline torch/utils/hipify/cuda_to_hip_mappings.py:613]: Undo this change.  This should be hipblas.

**Review by jeffdaily — CHANGES_REQUESTED** [review]

File: cmake/public/LoadHIP.cmake [hunk 2]
```diff
@@ -1,24 +1,20 @@
...
+  set(HIPSPARSE_PATH ${ROCM_PATH})
+  set(RCCL_PATH ${ROCM_PATH})
+  set(ROCPRIM_PATH ${ROCM_PATH})
+  set(HIPCUB_PATH ${ROCM_PATH})
+  set(ROCTHRUST_PATH ${ROCM_PATH})
+  set(HIPSOLVER_PATH ${ROCM_PATH})
+  set(ROCTRACER_PATH ${ROCM_PATH})
 endif()
 
 # HCC_PATH
```
Comment (pruthvistony) [inline cmake/public/LoadHIP.cmake:20]: This section can also be removed till line 145.

File: cmake/public/LoadHIP.cmake [hunk 3]
```diff
@@ -1,144 +1,20 @@
...
-  set(ROCTRACER_PATH $ENV{ROCTRACER_PATH})
-endif()
-
-# MAGMA_HOME
-if(NOT DEFINED ENV{MAGMA_HOME})
-  set(MAGMA_HOME ${ROCM_PATH}/magma)
-  set(ENV{MAGMA_HOME} ${ROCM_PATH}/magma)
-else()
-  set(MAGMA_HOME $ENV{MAGMA_HOME})
+if(ROCM_VERSION_DEV VERSION_GREATER_EQUAL "6.0.0")
```
Comment (jeffdaily) [inline cmake/public/LoadHIP.cmake:3]: ROCM_VERSION_DEV isn't defined until later in this file.

**Review by jithunnair-amd — APPROVED** [review]

**Review by pruthvistony — APPROVED** [review]

**Comment by pruthvistony** [comment]
@malfet ,
Can you please help review this PR.

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**